asb-ballerina/sender.bal

```diff
@@ -71,61 +78,61 @@ public isolated client class MessageSender {
     # + scheduledEnqueueTime - Datetime at which the message should appear in the Service Bus queue or topic
     # + return - The sequence number of the scheduled message which can be used to cancel the scheduling of the message
     isolated remote function schedule(@display {label: "Message Record or Payload"} Message message,
-            time:Civil scheduledEnqueueTime) returns int|error {
+            time:Civil scheduledEnqueueTime) returns int|Error {
         message.body = serializeToByteArray(message.body);
-        return schedule(self.senderHandle, message, scheduledEnqueueTime);
+        return schedule(self, message, scheduledEnqueueTime);
     }
 
     # Cancels the enqueuing of a scheduled message, if they are not already enqueued.
     #
     # + sequenceNumber - The sequence number of the message to cancel
     # + return - If the message could not be cancelled
-    isolated remote function cancel(@display {label: "Sequence Number"} int sequenceNumber) returns error? {
-        return cancel(self.senderHandle, sequenceNumber);
+    isolated remote function cancel(@display {label: "Sequence Number"} int sequenceNumber) returns Error? {
+        return cancel(self, sequenceNumber);
     }
 
     # Send batch of messages to queue or topic.
     #
     # + messageBatch - Azure service bus batch message representation (`asb:MessageBatch` record)
-    # + return - An error if failed to send message or else `()`
+    # + return - An `asb:Error` if failed to send message or else `()`
     @display {label: "Send Batch Message"}
-    isolated remote function sendBatch(@display {label: "Message Batch"} MessageBatch messageBatch) returns error? {
+    isolated remote function sendBatch(@display {label: "Message Batch"} MessageBatch messageBatch) returns Error? {
         foreach Message message in messageBatch.messages {
             message.body = serializeToByteArray(message.body);
         }
-        return sendBatch(self.senderHandle, messageBatch);
+        return sendBatch(self, messageBatch);
```

Comment:
This is a logic change right? Now we pass the whole object to the java side?

Reply:
Yeah this is done in order to make the sender implementation consistent with the receiver implementation (receiver is also implemented so that the native functions receive the whole client object itself so it has access to all the state variables)